drivers/serial/uart_psoc6.c

```diff
@@ -147,38 +138,31 @@ static const struct uart_driver_api uart_psoc6_driver_api = {
 	.poll_out = uart_psoc6_poll_out,
 };
 
-#ifdef CONFIG_UART_PSOC6_UART_5
-static const struct cypress_psoc6_config cypress_psoc6_uart5_config = {
-	.base = (CySCB_Type *)DT_REG_ADDR(DT_NODELABEL(uart5)),
-	.port = P5_0_PORT,
-	.rx_num = P5_0_NUM,
-	.tx_num = P5_1_NUM,
-	.rx_val = P5_0_SCB5_UART_RX,
-	.tx_val = P5_1_SCB5_UART_TX,
-	.scb_clock = PCLK_SCB5_CLOCK,
-};
-
-DEVICE_DT_DEFINE(DT_NODELABEL(uart5),
-			uart_psoc6_init, device_pm_control_nop, NULL,
-			&cypress_psoc6_uart5_config,
-			PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
-			(void *)&uart_psoc6_driver_api);
-#endif /* CONFIG_UART_PSOC6_UART_5 */
-
-#ifdef CONFIG_UART_PSOC6_UART_6
-static const struct cypress_psoc6_config cypress_psoc6_uart6_config = {
-	.base = (CySCB_Type *)DT_REG_ADDR(DT_NODELABEL(uart6)),
-	.port = P12_0_PORT,
-	.rx_num = P12_0_NUM,
-	.tx_num = P12_1_NUM,
-	.rx_val = P12_0_SCB6_UART_RX,
-	.tx_val = P12_1_SCB6_UART_TX,
-	.scb_clock = PCLK_SCB6_CLOCK,
-};
-
-DEVICE_DT_DEFINE(DT_NODELABEL(uart6),
-			uart_psoc6_init, device_pm_control_nop, NULL,
-			&cypress_psoc6_uart6_config,
-			PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
-			(void *)&uart_psoc6_driver_api);
-#endif /* CONFIG_UART_PSOC6_UART_6 */
+#define CY_PSOC6_UART_DECLARE_CFG(n, IRQ_FUNC_INIT)			\
+	static const struct cypress_psoc6_config cy_psoc6_uart##n##_config = { \
+		.base = (CySCB_Type *)DT_INST_REG_ADDR(n),		\
+		.periph_id = DT_INST_PROP(n, peripheral_id),		\
+									\
+		.num_pins = CY_PSOC6_DT_INST_NUM_PINS(n),		\
+		.pins = CY_PSOC6_DT_INST_PINS(n)			\
+	};
+
+#define CY_PSOC6_UART_IRQ_CFG_FUNC_INIT
+#define CY_PSOC6_UART_DECL_CONF(n)					\
+	CY_PSOC6_UART_DECLARE_CFG(n, CY_PSOC6_UART_IRQ_CFG_FUNC_INIT)
+#define CY_PSOC6_UART_DECL_DATA(n)
+#define CY_PSOC6_UART_DECL_DATA_PTR(n) NULL
```

Comment:
Let's maybe remove macros that are not needed now. They can be always added later. The less code to read and analyze the better.

Reply:
yeah old code.